docs/respository-configuration-cli.md

```diff
@@ -0,0 +1,383 @@
+[project]
+name = "demo-project"
+r_version = "4.4"
+repositories = [
+    { alias = "cran", url = "https://cran.r-project.org/" },
+    { alias = "bioc-data", url = "https://bioconductor.org/packages/3.18/data/annotation", force_source = true },
+]
+dependencies = [
+    "dplyr",
+    "ggplot2",
+]
+```
+
+## Example 10: Error scenarios
+
+### Duplicate alias error
+
+**Command:**
+```bash
+./target/release/rv configure repository --config-file test-example/rproject.toml --alias "cran" --url "https://packagemanager.posit.co/cran/2024-11-01" --first
+```
+
+**Output:**
+```
+Repository with alias 'cran' already exists
+```
+
+### Invalid URL error
+
+**Command:**
+```bash
+./target/release/rv configure repository --config-file test-example/rproject.toml --alias "invalid" --url "not-a-valid-url" --first
+```
+
+**Output:**
+```
+relative URL without a base
+```
+
+## Additional JSON Examples
+
+### JSON output for remove operation
+**Command:**
+```bash
+./target/release/rv --json configure repository --config-file test-example/rproject.toml --remove "cran"
+```
+
+**Output:**
+```json
+{
+  "operation": "remove",
+  "alias": "cran",
+  "url": null,
+  "success": true,
+  "message": "Repository removed successfully"
+}
+```
+
+### JSON output for replace operation
+**Command:**
+```bash
+./target/release/rv --json configure repository --config-file test-example/rproject.toml --alias "cran-updated" --url "https://packagemanager.posit.co/cran/latest" --replace "cran"
+```
+
+**Output:**
+```json
+{
+  "operation": "replace",
+  "alias": "cran-updated",
+  "url": "https://packagemanager.posit.co/cran/latest",
+  "success": true,
+  "message": "Repository replaced successfully"
+}
+```
+
+### JSON output for clear operation
+**Command:**
+```bash
+./target/release/rv --json configure repository --config-file test-example/rproject.toml --clear
+```
+
+**Output:**
+```json
+{
+  "operation": "clear",
+  "alias": null,
+  "url": null,
+  "success": true,
+  "message": "All repositories cleared"
+}
+```
+
+## Summary
+
+The `rv configure repository` command provides comprehensive repository management capabilities:
+
+1. **Positioning Control**: Add repositories exactly where needed with `--first`, `--last`, `--before`, and `--after` flags
+2. **Repository Management**: Replace existing repositories with `--replace`, remove specific ones with `--remove`, or clear all with `--clear`
+3. **Advanced Options**: Enable source compilation with `--force-source` flag
+4. **JSON Integration**: Comprehensive JSON output support for programmatic usage with `--json` flag, including operation type, alias, URL, success status, and descriptive messages
+5. **Error Handling**: Clear error messages for duplicate aliases, invalid URLs, and missing references
+6. **TOML Preservation**: Maintains file formatting and structure using `toml_edit`
+
+All commands preserve existing project structure and dependencies while only modifying the repositories section as requested.
```

Comment:
Do we actually need that file? I don't think a human will read that much for a simple command